### DISCUSSION TITLE
How do I open a website using a link not from a menu without components?

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#43629] [Q&A]
Please tell me if there is such an opportunity to open a website with an empty component, using a well-known link that was not created for the menu.
Ajax calls work without created links. but methods are called in modules and plugins. For several years, I have had a need from Joomla to open a page with design and layout by clicking on the link, as well as receiving content from the module or plugin function by clicking on the link. At the same time, so that only those modules for which there are no conditions for menu items are displayed.
Is this possible?

There is such a need to view the basket, the list of deferred materials, or view the invoice or sensor indicators

**Comment by Fedik** [comment]
Sorry, but your question is not very clear. 

When you need a custom link for custom view, then you probably can create a custom view in your component, and then use in menu.

**Reply by korenevskiy** [reply]
You're right, I can do this in the component. But I can't do it in a module or in a plugin.  As a developer, I write plug-in fields, a field indicating the price. Next, the plugin generates an order, and must send it to the shopping cart page.  The plugin turns content articles into products. Therefore, after selecting prices and quantities, the shopping cart or order confirmation page should open. A link to such a page is not required in the menu. Therefore, I need to open a page without a component, with information output from the plugin function.

**Reply by Fedik** [reply]
> Therefore, after selecting prices and quantities, the shopping cart or order confirmation page should open

For this you better do a litle component view.

Also, it is better when the data is managed by component (it is one of main job of the components).
With plugin you connect your component with joomla content.
And with module you display a litle piece of data from you component (it like a widgets). Which complement the component, but not replace it.
